plugins/TimingFanoutController.cpp

```diff
@@ -41,20 +42,21 @@ TimingFanoutController::TimingFanoutController(const std::string& name)
   register_command("fanout_io_reset", &TimingFanoutController::do_fanout_io_reset);
   register_command("fanout_print_status", &TimingFanoutController::do_fanout_print_status);
   register_command("fanout_endpoint_enable", &TimingFanoutController::do_fanout_endpoint_enable);
-  register_command("fanout_endpoint_reset", &TimingFanoutController::do_fanout_endpoint_enable);
+  register_command("fanout_endpoint_reset", &TimingFanoutController::do_fanout_endpoint_reset);
 }
 
 void
 TimingFanoutController::do_configure(const nlohmann::json& data)
 {
-  auto conf = data.get<timingfanoutcontroller::ConfParams>();
-  if (conf.device.empty())
+  auto mdal = m_params->module<dal::TimingFanoutController>(get_name()); 
+  if (mdal->get_device_str().empty())
   {
     throw UHALDeviceNameIssue(ERS_HERE, "Device name should not be empty");
   }
-  m_timing_device = conf.device;
-  m_hardware_state_recovery_enabled = conf.hardware_state_recovery_enabled;
-  m_timing_session_name = conf.timing_session_name;
+  
+  m_timing_device = mdal->get_device_str();
+  m_hardware_state_recovery_enabled = mdal->get_hardware_state_recovery_enabled();
+  m_timing_session_name = mdal->get_timing_session_name();
   m_device_ready_timeout = std::chrono::milliseconds(20000);
```

Comment:
I couldn't seem to find an mdal function for this, not sure if it is necessary or if the timeout can be handled as is?

Reply:
Surely if you need this value, you should define an attribute in the OKS schema, then there will be a getter function.